components/org.apache.stratos.rest.endpoint/src/main/java/org/apache/stratos/rest/endpoint/api/StratosApiV41Utils.java

```diff
@@ -1396,7 +1396,8 @@ public static void addApplication(ApplicationBean appDefinition, ConfigurationCo
                     usedCartridgeGroups.toArray(new String[usedCartridgeGroups.size()]));
 
         } catch (AutoscalerServiceApplicationDefinitionExceptionException e) {
-            throw new RestAPIException(e);
+            String message = e.getFaultMessage().getApplicationDefinitionException().getMessage();
```

Comment:
I think we may not need to handle null here, because if a NPE is raised it means that an error message raised by the Autoscaler service is not sent back to the client properly. If so we need to fix that problem. If we add a null check it would return a blank error message to the client.